### PR TITLE
Fixing looping fire sound and unauthenticated SetIntroCinematicMode packet

### PR DESCRIPTION
**Pull request by Jannify**
Tested with:
- [x] Skipping single player intro
- [x] Playing single player intro
- [x] Playing duo intro
- [x] Joining after intro played/second joining
- [x] Joining with creative mode
- [x] Joined with DEBUG build option

### CODE REVIEW

File: NitroxServer/Communication/Packets/Processors/SetIntroCinematicModeProcessor.cs [hunk 1]
```diff
@@ -26,6 +26,7 @@ public override void Process(SetIntroCinematicMode packet, Player player)
         packet.PartnerId = null; // Resetting incoming packets just to be safe we don't relay any PartnerId. Server has only authority.
         player.PlayerContext.IntroCinematicMode = packet.Mode;
         playerManager.SendPacketToOtherPlayers(packet, player);
+        Log.Debug($"Set IntroCinematicMode to {packet.Mode} for {player.PlayerContext.PlayerName}");
```
Comment (Coding-Hen) [inline NitroxServer/Communication/Packets/Processors/SetIntroCinematicModeProcessor.cs:29]: Do we want to keep this log or was this just used for testing?

File: NitroxPatcher/Patches/Dynamic/uGUI_SceneIntro_HandleInput_Patch.cs [hunk 2]
```diff
@@ -1,22 +1,72 @@
...
 {
-    private static readonly MethodInfo TARGET_METHOD = Reflect.Method((uGUI_SceneIntro t) => ((IInputHandler) t).HandleInput());
+    private static readonly MethodInfo TARGET_METHOD = Reflect.Method((uGUI_SceneIntro t) => ((IInputHandler)t).HandleInput());
 
-    public static void Postfix(uGUI_SceneIntro __instance, ref bool __result)
+    public static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
     {
-        if (!__result)
+        return new CodeMatcher(instructions)
+               // Remove uGUI_SceneIntro.Stop() and use our HandleButtonHeld feedback as return value
```
Comment (Coding-Hen) [inline NitroxPatcher/Patches/Dynamic/uGUI_SceneIntro_HandleInput_Patch.cs:20]: I like that we add this comment to say what it is actually trying to achieve. Do we want to follow the style we have seen in others with a bit more of a format? Just thinking then it is the same across other prs. Happy to not as I think this is clear but might be more beneficial for others

Something like 
Line before
Line - removed
New line -inserted
Existing line after

File: NitroxServer/Communication/Packets/Processors/SetIntroCinematicModeProcessor.cs (hunk 1 above)
Comment (Jannify) [reply NitroxServer/Communication/Packets/Processors/SetIntroCinematicModeProcessor.cs:29]: I want to keep this log. It's very helpful when something breaks to know the state the server had of the modes

**Review by Coding-Hen — APPROVED** [review]
LGTMCW

**Review by dartasen — APPROVED** [review]
Works fine IG

Note aside it's really disturbing to not have fire/extinguisher synced at the end of the cinematic 🧑‍🚒

**Pull request merged**